i6_models/parts/frontend/vgg.py

```diff
@@ -0,0 +1,354 @@
+
+        pool_stride = model_cfg.pool_stride
+        self.time_red = pool_stride[0] if isinstance(pool_stride, tuple) else pool_stride
+
+        self.include_linear_layer = True if model_cfg.linear_output_dim is not None else False
+
+        self.conv1 = nn.Conv2d(
+            in_channels=1,
+            out_channels=model_cfg.conv1_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pool = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_kernel_size,
+            stride=model_cfg.pool_stride,
+            padding=pool_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.conv1_channels,
+            out_channels=model_cfg.conv2_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            stride=model_cfg.conv2_stride,
+            padding=conv_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.conv2_channels,
+            out_channels=model_cfg.conv3_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.conv3_channels,
+            out_channels=model_cfg.conv4_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.activation = model_cfg.activation
+        if self.include_linear_layer:
+            self.linear = nn.Linear(
+                in_features=model_cfg.linear_input_dim,
+                out_features=model_cfg.linear_output_dim,
+                bias=True,
+            )
+
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        T might be reduced to T'' depending on the stride of the layers
+
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: masking tensor of shape [B,T], contains length information of the sequences
+        :return: torch.Tensor of shape [B,T',F']
+        """
+        tensor = tensor[:, None, :, :]  # [B,C=1,T,F]
+
+        tensor = self.conv1(tensor)
+        tensor = self.activation(tensor)
+        tensor = self.pool(tensor)  # [B,C,T',F]
+
+        tensor = self.conv2(tensor)  # [B,C,T",F]
+        tensor = self.activation(tensor)
+
+        tensor = self.conv3(tensor)
+        tensor = self.activation(tensor)
+
+        tensor = self.conv4(tensor)
+
+        tensor = torch.transpose(tensor, 1, 2)  # transpose to [B,T",C,F]
+        tensor = torch.flatten(tensor, start_dim=2, end_dim=-1)  # [B,T",C*F]
+
+        if self.include_linear_layer:
+            tensor = self.linear(tensor)
+
+        return tensor, sequence_mask
+
+
+def _get_padding(input_size: Union[int, Tuple[int, ...]]) -> Union[int, Tuple[int, ...]]:
+    """
+    get padding in order to not reduce the time dimension
+    :param input_size:
```

Comment:
I think there always should be an empty line between the description and the params.
```suggestion

    :param input_size:
```